src/authn/ccloudProvider.ts

```diff
@@ -129,27 +129,20 @@ export class ConfluentCloudAuthProvider implements vscode.AuthenticationProvider
       throw new Error("Failed to create new connection. Please try again.");
     }
 
-    try {
-      // this will block until we handle the URI event or the user cancels
-      await this.browserAuthFlow(signInUri);
-    } catch (e) {
-      if (e instanceof Error) {
-        vscode.window.showErrorMessage(
-          `Error during Confluent Cloud sign-in process: ${e.message}`,
-        );
-        logError(
-          e,
-          "browser sign-in flow",
-          {
-            authStatus: connection.status.authentication.status,
-            connectionId: connection.id,
-          },
-          true,
-        );
-      }
-      // this won't re-notify the user of the error, so no issue with re-throwing while showing the
-      // error notification above (if it exists)
-      throw e;
+    // this will block until we handle the URI event or the user cancels
+    const success: boolean | undefined = await this.browserAuthFlow(signInUri);
+    if (success === undefined) {
+      // user cancelled the operation
+      logger.debug("createSession() user cancelled the operation");
+      return Promise.reject(new Error("User cancelled the authentication flow."));
+    }
+    if (!success) {
+      const authFailedMsg = `Confluent Cloud authentication failed. See browser for details.`;
+      vscode.window.showErrorMessage(authFailedMsg);
+      logUsage(UserEvent.CCloudAuthentication, {
+        status: "authentication failed",
+      });
```

Comment:
Also adding a new telemetry event here so we can get better metrics on how often users run into CCloud authentication failure situations